packages/x-components/src/x-modules/query-suggestions/components/query-suggestion.vue

```diff
@@ -32,37 +30,47 @@
    *
    * @public
    */
-  @Component({
+  export default defineComponent({
+    name: 'QuerySuggestion',
     components: { BaseSuggestion },
-    mixins: [xComponentMixin(querySuggestionsXModule)]
-  })
-  export default class QuerySuggestion extends Vue {
-    /**
-     * The normalized query of the query-suggestions module.
-     *
-     * @internal
-     */
-    @Getter('querySuggestions', 'normalizedQuery')
-    public query!: string;
-
-    /**
-     * The suggestion to render.
-     *
-     * @public
-     */
-    @Prop({ required: true })
-    protected suggestion!: Suggestion;
-
-    /**
-     * Emits {@link QuerySuggestionsXEvents.UserSelectedAQuerySuggestion} with the suggestion as
-     * payload when selecting the query suggestion.
-     *
-     * @internal
-     */
-    protected event: Partial<XEventsTypes> = {
-      UserSelectedAQuerySuggestion: this.suggestion
-    };
-  }
+    xModule: querySuggestionsXModule.name,
+    props: {
+      /**
+       * The suggestion to render.
+       *
+       * @public
+       */
+      suggestion: {
+        type: Object as PropType<Suggestion>,
+        required: true
+      }
+    },
+    setup: function (props) {
+      useRegisterXModule(querySuggestionsXModule);
+
+      /**
+       * The normalized query of the query-suggestions module.
+       *
+       * @internal
+       */
+      const query = useGetter('querySuggestions', ['normalizedQuery']).normalizedQuery;
+
+      /**
+       * Emits {@link QuerySuggestionsXEvents.UserSelectedAQuerySuggestion} with the suggestion as
+       * payload when selecting the query suggestion.
+       *
+       * @internal
+       */
+      const event = {
+        UserSelectedAQuerySuggestion: props.suggestion
+      };
```

Comment:
```suggestion
      const suggestionSelectedEvents = {
        UserSelectedAQuerySuggestion: props.suggestion
      };
```
so later on the template v-bind could be cleaner:

`v-bind="{ query, suggestion, suggestionSelectedEvents }"`